### ISSUE TITLE
Cannot call custom function in 2.1.0

### ISSUE THREAD

**Issue by martinkirch** [savonet/liquidsoap#2560]
Using Liquidsoap 2.1.0 built with opam, on Fedora36.

I'm using a script with a source creation function (shortened here):
```
def recorded_show(name)
  show_folder = "#{SHOWS}/#{name}"
  playlist(reload_mode="watch", mode="normal", id=name, show_folder)
end
```

When updating to 2.1, liquidsoap fails to start when this function is called:
```
example = recorded_show("Example")

Error 4: Undefined variable recorded_show
```

Starting the function with ` def recorded_show(name) =` didn't work either.

The same script worked fined up to v2.0.7

**Comment by vitoyucepi** [comment]
Hi, @martinkirch.

It seems there's a bug in the parser associated with `rec` keyword.
Liquidsoap thinks that `def recf()` is definition of recursive function `f()` and is equivalent of `def rec f()`.
```ruby
SHOWS='/tmp'

def recorded_show(name)
  show_folder = "#{SHOWS}/#{name}"
  playlist(reload_mode="watch", mode="normal", id=name, show_folder)
end

example = orded_show("Example")
output.dummy(mksafe(example))
```

**Comment by vitoyucepi** [comment]
This issue exists in *main* too.
```ruby
def reca()
()
end
a()
```
https://www.liquidsoap.info/try/

**Comment by toots** [comment]
Woof. Thanks for reporting and for figuring it out y'all! This is fixed now in `main` and in `rolling-release-v2.1.x`. This probably a good candidate for a `v2.1.x` release soon-ish and I believe this may randomly hit others.
